client/python/armada_client/client.py

```diff
@@ -0,0 +1,144 @@
+        request = submit_pb2.JobCancelRequest(
+            queue=queue, job_id=job_id, job_set_id=job_set_id
+        )
+        response = self.submit_stub.CancelJobs(request)
+        return response
+
+    def reprioritize_jobs(
+        self, new_priority, job_ids=None, job_set_id=None, queue=None
+    ):
+        """Implementation of ReprioritizeJobs rpc function"""
+
+        request = submit_pb2.JobReprioritizeRequest(
+            job_ids=job_ids,
+            job_set_id=job_set_id,
+            queue=queue,
+            new_priority=new_priority,
+        )
+        response = self.submit_stub.ReprioritizeJobs(request)
+        return response
+
+    def create_queue(self, name, **queue_params):
+        """Implementation of CreateQueue rpc function"""
+        request = submit_pb2.Queue(name=name, **queue_params)
+        response = self.submit_stub.CreateQueue(request)
+        return response
+
+    def update_queue(self, name, **queue_params):
+        """Implementation of UpdateQueue rpc function"""
+        request = submit_pb2.Queue(name=name, **queue_params)
+        response = self.submit_stub.UpdateQueue(request)
+        return response
+
+    def delete_queue(self, name):
+        """Implementation of DeleteQueue rpc function"""
+        request = submit_pb2.QueueDeleteRequest(name=name)
+        response = self.submit_stub.DeleteQueue(request)
+        return response
+
+    def get_queue(self, name):
+        """Impl of GetQueue"""
+        request = submit_pb2.QueueGetRequest(name=name)
+        response = self.submit_stub.GetQueue(request)
+        return response
+
+    def get_queue_info(self, name):
+        """Impl of GetQueueInfo"""
+        request = submit_pb2.QueueInfoRequest(name=name)
+        response = self.submit_stub.GetQueueInfo(request)
+        return response
+
+    def watch_events(self, on_event, queue, job_set_id, from_message_id=None):
+        """Impl of WatchEvents"""
+        jsr = event_pb2.JobSetRequest(
+            queue=queue,
+            id=job_set_id,
+            from_message_id=from_message_id,
+            watch=True,
+            errorIfMissing=True,
+        )
+        event_stream = self.event_stub.GetJobSetEvents(jsr)
+
+        def event_counter():
+            try:
+                nonlocal event_stream
+                for event in event_stream:
+                    on_event(event)
+            except grpc._channel._MultiThreadedRendezvous as error:  # pylint: disable=protected-access
+                if error.code() == grpc.StatusCode.CANCELLED:
+                    pass
+                # process cancelled status
+                elif (
+                    error.code() == grpc.StatusCode.UNAVAILABLE
+                    and "Connection reset by peer" in error.details()
+                ):
+                    pass
+                # process unavailable status
+                else:
+                    raise
+
+        self.executor.submit(event_counter)
```

Comment:
I think functols.partial is nicer than using nonlocal to pass in variables. Makes it more clear what gets passed in.